tidy3d/components/geometry.py

```diff
@@ -957,15 +991,35 @@ def set_length(cls, val, values):
 
     @pydantic.validator("vertices", always=True)
     def correct_shape(cls, val):
-        """makes sure vertices is correct shape if numpy array"""
-        if isinstance(val, np.ndarray):
-            shape = val.shape
-            if len(shape) != 2 or shape[1] != 2:
-                raise SetupError(
-                    "PolySlab.vertices must be a 2 dimensional array shaped (N, 2).  "
-                    f"Given array with shape of {shape}."
-                )
-        return val
+        """makes sure vertices size is correct. Remove duplicate
+        neighbouring vertices, orient vertices in CCW direction.
+        make sure no intersecting edges.
+        """
+
+        val_np = PolySlab.vertices_to_array(val)
+        shape = val_np.shape
+
+        # overall shape of vertices
+        if len(shape) != 2 or shape[1] != 2:
+            raise SetupError(
+                "PolySlab.vertices must be a 2 dimensional array shaped (N, 2).  "
+                f"Given array with shape of {shape}."
+            )
+
+        # remove duplicate neighbouring vertices, and orient in CCW direction
+        vertices = PolySlab._proper_vertices(val)
```

Comment:
this gets applied to `val`, not `val_np` and then it seems `vertices` is never used besides validation (the `val_np` is returned). Is this expected?

Reply:
vertices is used in https://github.com/flexcompute/tidy3d/blob/a71560c1b26fd860edf01f577c56c1d48f6d9ee6/tidy3d/components/geometry.py#L1010 and https://github.com/flexcompute/tidy3d/blob/a71560c1b26fd860edf01f577c56c1d48f6d9ee6/tidy3d/components/geometry.py#L1017

Reply:
it is expected. They are only used in validation.